app/controllers/organizations_controller.rb

```diff
@@ -80,12 +80,12 @@ def create
   private
 
   def find_organization
-    if current_user && user_is_admin?
-      @organization = Organization.find params[:id]
+    @organization = Organization.find params[:id]
+    if current_user && user_is_admin? || (current_user && current_user.organization_id == @organization.id)
```

Comment:
This doesn't seem to me like it would do the job, can you please write three tests to confirm this:
- Test to visit the unpublished organization page when the user is not logged in
- Test to visit the unpublished organization page as an admin
- Test to visit the unpublished organization page as an owner of the organization